v-next/hardhat-utils/src/errors/bytecode.ts

```diff
@@ -0,0 +1,48 @@
+import { CustomError } from "../error.js";
+
+export class InvalidLibraryAddressError extends CustomError {
+  constructor(formattedLibraries: string) {
```

Comment:
Is there an argument here for moving the formatting code that precedes the throwing of these errors into the custom errors themselves?

So passing a structured type in the constructor parameter and converting it to a string here for the message rather than putting the formatting code at the call point?

Reply:
The idea behind this approach was to keep the error classes simple. However, your point about shifting the responsibility of formatting to the error classes is valid. I’ll update the code accordingly.